CruCentralCoast/Extensions/UIViewControllerExtensions.swift

```diff
@@ -48,4 +54,25 @@ public extension UIViewController {
         alertController.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
         self.present(alertController, animated: animated, completion: completion)
     }
+    
+    func showWebView(from url: String, withActivityIndicator: Bool = false, navigationDelegate: WKNavigationDelegate?) {
+        let webView = WKWebView()
+        if let url = URL(string: url) {
+            webView.load(URLRequest(url: url))
+        }
+        let vc = UIViewController()
+        vc.view = webView
+        webView.navigationDelegate = navigationDelegate
+        if (withActivityIndicator) {
+            let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
+            activityIndicator.hidesWhenStopped = true
+            activityIndicator.color = .gray
+            webView.addSubview(activityIndicator)
```

Comment:
Now I see that it is the navigation delegate's job to stop animating it, which makes sense

Reply:
Although, I wonder if it would be possible to assign the UIActivityIndicatorView to a variable in the navigation delegate so that later you don't have to find it by searching through the subviews?

Reply:
good idea. ill change some stuff